File: CSharp/DP.Domain/Samples/Factory/AbsDbFactory.cs
namespace DP.Domain.Samples.Factory
{
    public abstract class AbsDbFactory
    {
        public abstract DbContext Create();
    }

    public class DataMartDbFactory : AbsDbFactory
    {
        public override DbContext Create()
        {
            return new DbContext()
            {
                Server = "DataMart",
                ConnectionStr = "DataMart connection string"
            };
        }
    }

    public class HistoryDbFactory : AbsDbFactory
    {
        public override DbContext Create()
        {
            return new DbContext()
            {
                Server = "History",
                ConnectionStr = "History connection string"
            };
        }
    }

    public class OnlineDbFactory : AbsDbFactory
    {
        public override DbContext Create()
        {
            return new DbContext()
            {
                Server = "Online",
                ConnectionStr = "Online connection string"
            };
        }
    }
}

File: Python/Samples/Factory/AbsDbFactory.py
from abc import ABC, abstractmethod
from Factory_models import DbContext

class AbsDbFactory(ABC):
    @abstractmethod
    def create(self):
        """Return DbContext"""
        pass

class DataMartDbFactory(AbsDbFactory):
    def create(self):
        return DbContext(server = "DataMart",connectionStr = "DataMart connection string")

class HistoryDbFactory(AbsDbFactory):
    def create(self):
        return DbContext(server = "History",connectionStr = "History connection string")

class OnlineDbFactory(AbsDbFactory):
    def create(self):
        return DbContext(server = "Online",connectionStr = "Online connection string")